### DISCUSSION TITLE
Escape characters in config file?

### DISCUSSION THREAD

**Discussion by jpretori** [i3/i3#4541] [Q&A] [answered]
I've been using i3wm on Mint 20.1 for a few months now, and have been generally very happy.  Recently I tried to configure a compose key so that I can make characters like `é` without much trouble.

So I did some searching and found that this is the command I need: ```xmodmap -e "keycode 105 = Multi_key"``` and that it needs to be run every time my session starts.

This works in my terminal, whether I use single or double quotes.

So I went into ```~/.config/i3/config``` and added this line:

```exec --no-startup-id i3-msg 'xmodmap -e \\"keycode 105 = Multi_key\\"'```

Which doesn't work.  I tried using single `\` as the escape character, that doesn't work either.  The i3 manual doesn't seem to offer any hints on how to make this work.

What should I do in the i3 config file to execute a command on startup, if that command requires the use of single or double quotes?

**Comment by Airblader** [comment, marked as answer]
In general I'd recommend putting the command in a script instead and execute that instead of dealing with escaping it.

**Reply by dojero** [reply]
The xmodmap -e command does't last. But you can make the change permanent by creating an ~/.Xmodmap file (with the command xmodmap > ~/.Xmodmap). Then you can modify you .Xmodmap file with your keycode 105 change. Once you do, X will check the .Xmodmap file and do what it says. There's no need for i3 to be involved.

**Comment by jpretori** [comment]
Thanks!  For the sake of future people doing web searches and finding this thread, I discovered that running `sudo dpkg-reconfigure keyboard-configuration` leads one through a set of what looks like ncurses-based dialogs that asks for some keyboard setup preferences.  

This process appears to update the file `/etc/default/keyboard`.  In my particular case, it seems to have added a line `XKBOPTIONS="compose:ralt"` to that file.  It works now, even across reboots.
